mesonbuild/dependencies/boost.py

```diff
@@ -615,15 +611,19 @@ def filter_libraries(self, libs: T.List[BoostLibraryFile], lib_vers: str) -> T.L
         return libs
 
     def detect_libraries(self, libdir: Path) -> T.List[BoostLibraryFile]:
-        libs = []  # type: T.List[BoostLibraryFile]
+        libs = set()  # type: T.Set[Path]
         for i in libdir.iterdir():
-            if not i.is_file() or i.is_symlink():
+            if not i.is_file():
                 continue
             if not any([i.name.startswith(x) for x in ['libboost_', 'boost_']]):
                 continue
 
-            libs += [BoostLibraryFile(i)]
-        return [x for x in libs if x.is_boost()]  # Filter out no boost libraries
+            libs.add(i.resolve())
```

Comment:
When I suggested making it into a set from the beginning, I was kind of assuming the logical next step is to consolidate this. You no longer need multiple complex filtering passes, so...

```suggestion
            libs.add(BoostLibraryFile(i.resolve()))
```

And then you don't need to assign blibs and make the return line refer to a different variable. This would help reduce the size of the diff... :)

Reply:
I made the requested change